test/metabase/legacy_mbql/util_test.cljc

```diff
@@ -945,3 +946,73 @@
+                   [[:= [:field 1 nil] 6]  "Jun"]
+                   [[:= [:field 1 nil] 7]  "Jul"]
+                   [[:= [:field 1 nil] 8]  "Aug"]
+                   [[:= [:field 1 nil] 9]  "Sep"]
+                   [[:= [:field 1 nil] 10] "Oct"]
+                   [[:= [:field 1 nil] 11] "Nov"]
+                   [[:= [:field 1 nil] 12] "Dec"]]
+            {:default ""}]
+           (mbql.u/desugar-expression [:month-name [:field 1 nil]]))
+        "`month-name` should desugar to a `:case` clause with values for each month")
+  (t/is (= [:case [[[:= [:field 1 nil] 1] "Q1"]
+                   [[:= [:field 1 nil] 2] "Q2"]
+                   [[:= [:field 1 nil] 3] "Q3"]
+                   [[:= [:field 1 nil] 4] "Q4"]]
+            {:default ""}]
+           (mbql.u/desugar-expression [:quarter-name [:field 1 nil]]))
+        "`quarter-name` should desugar to a `:case` clause with values for each quarter")
+  (t/is (= [:case [[[:= [:field 1 nil] 1] "Sunday"]
+                   [[:= [:field 1 nil] 2] "Monday"]
+                   [[:= [:field 1 nil] 3] "Tuesday"]
+                   [[:= [:field 1 nil] 4] "Wednesday"]
+                   [[:= [:field 1 nil] 5] "Thursday"]
+                   [[:= [:field 1 nil] 6] "Friday"]
+                   [[:= [:field 1 nil] 7] "Saturday"]]
+            {:default ""}]
+           (mbql.u/desugar-expression [:day-name [:field 1 nil]]))
+        "`day-name` should desugar to a `:case` clause with values for each weekday"))
+
+#?(:clj
+   (t/deftest ^:synchronized desugar-month-quarter-day-name-i18n-test
+     (mt/with-locale "es"
+       (t/is (= [:case [[[:= [:field 1 nil] 1]  "ene"]
+                        [[:= [:field 1 nil] 2]  "feb"]
+                        [[:= [:field 1 nil] 3]  "mar"]
+                        [[:= [:field 1 nil] 4]  "abr"]
+                        [[:= [:field 1 nil] 5]  "may"]
+                        [[:= [:field 1 nil] 6]  "jun"]
+                        [[:= [:field 1 nil] 7]  "jul"]
+                        [[:= [:field 1 nil] 8]  "ago"]
+                        [[:= [:field 1 nil] 9]  "sept"]
+                        [[:= [:field 1 nil] 10] "oct"]
+                        [[:= [:field 1 nil] 11] "nov"]
+                        [[:= [:field 1 nil] 12] "dic"]]
+                 {:default ""}]
+                (mbql.u/desugar-expression [:month-name [:field 1 nil]]))
+             "`month-name` should desugar to a `:case` clause with values for each month")
+       (t/is (= [:case [[[:= [:field 1 nil] 1] "Q1"]
+                        [[:= [:field 1 nil] 2] "Q2"]
+                        [[:= [:field 1 nil] 3] "Q3"]
+                        [[:= [:field 1 nil] 4] "Q4"]]
+                 {:default ""}]
+                (mbql.u/desugar-expression [:quarter-name [:field 1 nil]]))
+             "`quarter-name` should desugar to a `:case` clause with values for each quarter")
+       (t/is (= [:case [[[:= [:field 1 nil] 1] "domingo"]
```

Comment:
OK, so it seems the locale doesn't take care of selecting the locale specific first day of the week. (Or does it change if we specify the country (es_ES vs es_AR)? If the choice is intentional, then we should probably document it.